src/applications/spatial.c

```diff
@@ -314,340 +477,382 @@ int starsh_ssdata_generate_el(STARSH_ssdata **data, STARSH_int count, ...)
-            break;
-        case STARSH_SPATIAL_EXP_SIMD:
-            *kernel = starsh_ssdata_block_exp_kernel_4d_simd;
-            break;
-        case STARSH_SPATIAL_SQREXP:
-            *kernel = starsh_ssdata_block_sqrexp_kernel_4d;
-            break;
-        case STARSH_SPATIAL_SQREXP_SIMD:
-            *kernel = starsh_ssdata_block_sqrexp_kernel_4d_simd;
-            break;
+	case STARSH_SPATIAL_EXP:
+	    *kernel = starsh_ssdata_block_exp_kernel_4d;
+	    break;
+	case STARSH_SPATIAL_EXP_SIMD:
+	    *kernel = starsh_ssdata_block_exp_kernel_4d_simd;
+	    break;
+	case STARSH_SPATIAL_SQREXP:
+	    *kernel = starsh_ssdata_block_sqrexp_kernel_4d;
+	    break;
+	case STARSH_SPATIAL_SQREXP_SIMD:
+	    *kernel = starsh_ssdata_block_sqrexp_kernel_4d_simd;
+	    break;
 #ifdef GSL
-        case STARSH_SPATIAL_MATERN:
-            *kernel = starsh_ssdata_block_matern_kernel_4d;
-            break;
-        case STARSH_SPATIAL_MATERN_SIMD:
-            *kernel = starsh_ssdata_block_matern_kernel_4d_simd;
-            break;
-        case STARSH_SPATIAL_MATERN2:
-            *kernel = starsh_ssdata_block_matern2_kernel_4d;
-            break;
-        case STARSH_SPATIAL_MATERN2_SIMD:
-            *kernel = starsh_ssdata_block_matern2_kernel_4d_simd;
-            break;
+	case STARSH_SPATIAL_MATERN:
+	    *kernel = starsh_ssdata_block_matern_kernel_4d;
+	    break;
+	case STARSH_SPATIAL_MATERN_SIMD:
+	    *kernel = starsh_ssdata_block_matern_kernel_4d_simd;
+	    break;
+	case STARSH_SPATIAL_MATERN2:
+	    *kernel = starsh_ssdata_block_matern2_kernel_4d;
+	    break;
+	case STARSH_SPATIAL_MATERN2_SIMD:
+	    *kernel = starsh_ssdata_block_matern2_kernel_4d_simd;
+	    break;
 #else
-        case STARSH_SPATIAL_MATERN:
-        case STARSH_SPATIAL_MATERN_SIMD:
-        case STARSH_SPATIAL_MATERN2:
-        case STARSH_SPATIAL_MATERN2_SIMD:
-            STARSH_ERROR("Matern kernel requires GSL library, which was "
-                    "not found");
-            return STARSH_WRONG_PARAMETER;
+	case STARSH_SPATIAL_MATERN:
+	case STARSH_SPATIAL_MATERN_SIMD:
+	case STARSH_SPATIAL_MATERN2:
+	case STARSH_SPATIAL_MATERN2_SIMD:
+	case STARSH_SPATIAL_PARSIMONIOUS_SIMD:
+	case STARSH_SPATIAL_PARSIMONIOUS2_SIMD:
+	    STARSH_ERROR("Matern kernel requires GSL library, which was "
+		    "not found");
+	    return STARSH_WRONG_PARAMETER;
 #endif
-        case STARSH_SPATIAL_EXP_GCD:
-        case STARSH_SPATIAL_SQREXP_GCD:
-        case STARSH_SPATIAL_MATERN_GCD:
-        case STARSH_SPATIAL_MATERN2_GCD:
-            STARSH_ERROR("GCD (spherical distance) can be used only for 2D "
-                    "problem");
-            break;
-        default:
-            STARSH_ERROR("Wrong type of kernel");
-            return STARSH_WRONG_PARAMETER;
+	case STARSH_SPATIAL_EXP_GCD:
+	case STARSH_SPATIAL_SQREXP_GCD:
+	case STARSH_SPATIAL_MATERN_GCD:
+	case STARSH_SPATIAL_MATERN2_GCD:
+	case STARSH_SPATIAL_PARSIMONIOUS_GCD:
```

Comment:
case STARSH_SPATIAL_PARSIMONIOUS shall return error here, since this kernel does not support generic value of dimensionality (ndim=4, although kernel is meant only for 2d case).